services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -26,6 +26,73 @@ export const [_, PanicOverview] = createStore({ error: '' });
 export const useApplicationNames = () =>
     useOverview(({ applications }) => Object.keys(applications).sort((a, b) => a.localeCompare(b)));
 
+// return all environment locks
+export const useEnvironmentLocks = () =>
+    useOverview(({ environments }) => {
+        const locks = Object.values(environments)
+            .map((environment) =>
+                Object.values(environment.locks).map(
+                    (lockInfo) =>
+                        ({
+                            date: lockInfo.createdAt,
+                            environment: environment.name,
+                            lockId: lockInfo.lockId,
+                            message: lockInfo.message,
+                            authorName: lockInfo.createdBy?.name,
+                            authorEmail: lockInfo.createdBy?.email,
+                        } as DisplayLock)
+                )
+            )
+            .filter((displayLock) => displayLock.length !== 0)
+            .flat();
+        return sortLocks(locks);
+    });
+
+// return all applications locks
+export const useApplicationLocks = () =>
+    useOverview(({ environments }) => {
+        const locks = Object.values(environments)
+            .map((environment) =>
+                Object.values(environment.applications)
+                    .map((application) =>
+                        Object.values(application.locks).map(
+                            (lockInfo) =>
+                                ({
+                                    date: lockInfo.createdAt,
+                                    application: application.name,
+                                    environment: environment.name,
+                                    lockId: lockInfo.lockId,
+                                    message: lockInfo.message,
+                                    authorName: lockInfo.createdBy?.name,
+                                    authorEmail: lockInfo.createdBy?.email,
+                                } as DisplayLock)
+                        )
+                    )
+                    .filter((displayLock) => displayLock.length !== 0)
+            )
+            .filter((displayLock) => displayLock.length !== 0)
+            .map((displayLock) => displayLock[0])
+            .flat();
+        return sortLocks(locks);
+    });
+
+const sortLocks = (displayLocks: DisplayLock[]) =>
```

Comment:
I'm really not feeling good about this function, it'stoo complex sorting, and it only sorts in ascending order, what about the descending order ?

Reply:
I agree with you, and It is already implemented. I allow that the locks to be sorted by ascending or descending order (first it orders by date, then environment, then application, and so on).